pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -417,52 +424,114 @@ def get_checksum_types(self, **kwargs):
 
         return repomd_checksum_types, primary_checksum_types
 
-    def test_unspecified_checksum_types(self):
-        """Sync and publish an RPM repository and verify the checksum."""
-        repomd_checksum_types, primary_checksum_types = self.get_checksum_types()
+    def test_on_demand_unspecified_checksum_types(self):
+        """Sync and publish an RPM repository and verify the checksum types."""
+        repomd_checksum_types, primary_checksum_types = self.get_checksum_types(
+            fixture_url=RPM_SHA512_FIXTURE_URL, policy="on_demand"
+        )
+
+        for repomd_type, repomd_checksum_type in repomd_checksum_types.items():
+            # hack to account for the fact that our sha5 and sha512 repos don't
+            # universally use sha5 and sha256 checksums for metadata :(
```

Comment:
I do not follow this comment, could you check if all shaXXX are correctly specified.

Did you mean to say that "our sha1 and sha512 repos don't universally use sha1 and sha512 checksums for metadata"? or md5 and sha512?

FWIW, there is the same comment in the next test.